sqlx-sqlite/src/connection/mod.rs

```diff
@@ -51,6 +53,11 @@ pub struct LockedSqliteHandle<'a> {
     pub(crate) guard: MutexGuard<'a, ConnectionState>,
 }
 
+/// Represents a callback handler that will be shared with the underlying sqlite3 connection.
+pub(crate) struct Handler(NonNull<dyn FnMut() -> bool>);
+unsafe impl Send for Handler {}
+unsafe impl Sync for Handler {}
```

Comment:
We shouldn't need to force this to be `Sync`, especially since we're not expecting the handler itself to be `Sync`.

Reply:
This was implemented because the compiler wanted `NonNull` to be `Send`, since `NonNull` is `!Send`. As we are taking care that the data is not aliased, we should be able to mark it as `Send`.